Document differences between this repo and upstream (Apache) yunikorn-core

richscott opened the pull request:
This PR simply documents the functional differences between this repository and the upstream repo (https://github.com/apache/yunikorn-core) from which it was forked.

Fixes https://github.com/G-Research/unicorn-history-server/issues/319

nikola-jokic committed:
Suggested changes

richscott committed:
Add start of doc to detail fork differences

Add a markdown document describing the differences between
this repository and the upstream Apache yunikorn-core repository.

Fixes https://github.com/G-Research/unicorn-history-server/issues/319

richscott committed:
Add start of doc to detail fork differences

Add a markdown document describing the differences between
this repository and the upstream Apache yunikorn-core repository.

Fixes https://github.com/G-Research/unicorn-history-server/issues/319

richscott committed:
Markdown touch-ups, fix typos.

richscott committed:
Markdown touch-ups, fix typos.

nikola-jokic committed:
Extract functionality

nikola-jokic committed:
Update files with license

nikola-jokic commented on UNICORN-CHANGES.md:
```diff
@@ -0,0 +1,48 @@
+- The `PartitionQueueDAOInfo` object now has a `ParentID *string` field, which contains the `ID`
+value of that queue's parent queue. If it is the root queue, this field will be *nil*.
+
+- The Allocation scheduler object now has two new methods for getting external DAO representations of 
```
I ended up exposing more methods so I can test any interface changes. I don't think we need to document these public methods to describe changes. It would be hard to maintain. The most important thing is that we extended the event stream and assigned IDs to objects where name was used. I think this should be just a high-level overview. Otherwise, we would have to document everything whenever we touch something.

CC @dave-gantenbein

richscott replied:
Okay, I remove mentions of the methods, and just left in the description of the various ID additions - let me know if you think more should be trimmed - commit 6324a26 .

richscott committed:
Set a node as schedulable after we get/set its partition

When adding a node, call SetSchedulable() on it after we've obtained its
partition and set its PartitionID member - this is so the Node add/set
event that is published will have its PartitionID populated.

Fixes https://github.com/G-Research/unicorn-history-server/issues/342

dave-gantenbein committed:
Merge pull request #9 from G-Research/rich/node-partion-id

Yunikorn Core: fix for missing partition_id on NODE ADD/SET events

nikola-jokic committed:
update go version and testify

richscott committed:
Merge pull request #8 from G-Research/nikola-jokic/sync-test-check

Nikola jokic/sync test check

richscott committed:
Merge branch 'rich/gr-differences-doc' of github.com:G-Research/yunikorn-core into rich/gr-differences-doc

richscott committed:
Remove mentions of new methods; keep data-stream object changes.

richscott commented:
Closing this PR, due to GH using a different upstream branch and showing many files changes (there is actually only one). A new PR will be opened.
